### ISSUE TITLE
.NET 6 support

### ISSUE THREAD

**Issue by leppie** [IronScheme/IronScheme#113]
![image](https://user-images.githubusercontent.com/127666/141499401-862b6028-b7a6-4d9c-95e9-13db88874184.png)

**Comment by leppie** [comment]
With stack: 
![image](https://user-images.githubusercontent.com/127666/141499662-f90e092c-e784-42df-8fb5-9d21572bcbb9.png)

**Comment by leppie** [comment]
Also Ctrl-C gives an error:
![image](https://user-images.githubusercontent.com/127666/141499778-12c39665-1f14-4807-818b-812e810b7e81.png)

**Comment by leppie** [comment]
Seems they broke something badly.... 
![image](https://user-images.githubusercontent.com/127666/141506435-ddd56da2-9616-4c0d-8d3f-d9956af6ad15.png)

**Comment by bcalco** [comment]
Wow, seems like the type system is seeing double where there are no doubles. Literally everything is the same, except the runtime thinks A is not B, all other thing literally being equal.

**Comment by leppie** [comment]
@bcalco  That was just a build artefact. 

Please test with this release https://github.com/IronScheme/IronScheme/releases/tag/1.0.252

**Comment by bcalco** [comment]
@leppie Nope, same stack trace, as far as I can tell.

```
C:\Users\adsmactf01>adb repl
AccelerateDB Data REPL - 2021.4 © 2021 Apex Data Solutions, LLC. (.NET 6.0.0-rtm.21522.10 - 64-bit)
Powered by Visual Scheme for Applications (VSA) 2021.4 © 2021 Apex Data Solutions, LLC.

> (debug-mode? #t)
> (display-stacktrace #t)
> (import (unit-testing))
> (check that
.   (+ 2 2)
.  from "the REPL"
.  is numerically equal to 4
.  because "it would really suck if basic math didn't work.")
Unhandled exception during evaluation:
&who: #f
&message: "The method or operation is not implemented."
&syntax:
  form: #f
  subform: #f
&stacktrace
  [1] "ironscheme.exceptions::raise(obj)"
  [2] "ironscheme.exceptions::syntax-error(who, msg, form, subform)"
  [3] "IronScheme.Runtime.Glue.SyntaxError(who, message, form, subform)"
  [4] "IronScheme.Runtime.Builtins.CompileCore(expr)"
  [5] "psyntax.expander::library-expander##anon#1#2$2631()"
  [6] "psyntax.library-manager::invoke-library(lib)"
  [7] "ironscheme.lists::for-each(f, arg1)"
  [8] "psyntax.library-manager::invoke-library(lib)"
  [9] "ironscheme.lists::for-each(f, arg1)"
  [10] "psyntax.library-manager::invoke-library(lib)"
  [11] "ironscheme.lists::for-each(f, arg1)"
  [12] "psyntax.expander::eval(x, env)"
  [13] "psyntax.main::eval-top-level#anon#1#2#3#4#5$2698()"
  [14] "ironscheme.exceptions::dynamic-wind(in, proc, out)"
  [15] "psyntax.main::eval-top-level#anon#1#2$2696()"
  [16] "ironscheme.exceptions::dynamic-wind(in, proc, out)"
  [17] "psyntax.main::eval-top-level#anon#1$2692(k)"
  [18] "IronScheme.Runtime.Builtins.CallWithCurrentContinuation(fc1)"
  [19] "IronScheme.Runtime.R6RS.Exceptions.WithClrExceptionHandler(handler, thunk)"
  [20] "psyntax.main::eval-top-level(x)"
  [21] "IronScheme.Scripting.Hosting.CompiledCode.Evaluate(IScriptModule module)"
  [22] "IronScheme.Scripting.Hosting.ScriptEngine.ExecuteInteractiveCode(String code, IScriptModule module)"
  [23] "IronScheme.Scripting.Hosting.ScriptEngine.ExecuteCommand(String code, IScriptModule module)"
  [24] "IronScheme.Scripting.Shell.CommandLine.RunOneInteraction()"
  [25] "IronScheme.Scripting.Shell.CommandLine.TryInteractiveAction()"
  [26] "IronScheme.Scripting.Shell.CommandLine.RunInteractiveLoop()"
  [27] "IronScheme.Scripting.Shell.CommandLine.RunInteractive()"
  [28] "IronScheme.Scripting.Shell.CommandLine.Run()"
  [29] "IronScheme.Scripting.Shell.CommandLine.Run(IScriptEngine engine, IConsole console, ConsoleOptions options)"
  [30] "IronScheme.Scripting.Hosting.ConsoleHost.RunCommandLine(OptionsParser optionsParser)"
  [31] "IronScheme.Scripting.Hosting.ConsoleHost.ExecuteInternal()"
  [32] "IronScheme.Scripting.Hosting.ConsoleHost.Execute()"
  [33] "IronScheme.Scripting.Hosting.ConsoleHost.Run(String[] args)"
  [34] "AccelerateDB.Scripting.AccelerateDBConsoleHost.Execute(String[] argv) in \\\\Mac\\Home\\Documents\\GitHub\\Apex.Unify\\Source\\DomainActivate\\Common\\AccelerateDB\\Scripting.fs:line 36"
  [35] "AccelerateDB.Scripting.AccelerateDBRuntime.RunConsole(String[] args) in \\\\Mac\\Home\\Documents\\GitHub\\Apex.Unify\\Source\\DomainActivate\\Common\\AccelerateDB\\Scripting.fs:line 50"
  [36] "AccelerateDB.Scripting.REPL.runAccelerateDBConsole(String[] args) in \\\\Mac\\Home\\Documents\\GitHub\\Apex.Unify\\Source\\DomainActivate\\Common\\AccelerateDB\\Scripting.fs:line 57"
  [37] "AccelerateDB.Console.Commands.REPL.start(ParseResults`1 args) in \\\\Mac\\Home\\Documents\\GitHub\\Apex.Unify\\Source\\DomainActivate\\Common\\adb\\Commands.fs:line 289"
  [38] "AccelerateDB.Console.App.start(String name, String banner, String[] argv) in \\\\Mac\\Home\\Documents\\GitHub\\Apex.Unify\\Source\\DomainActivate\\Common\\adb\\App.fs:line 68"
  [39] "AccelerateDB.Console.Main.main(String[] argv) in \\\\Mac\\Home\\Documents\\GitHub\\Apex.Unify\\Source\\DomainActivate\\Common\\adb\\Program.fs:line 24"
```

**Comment by leppie** [comment]
You might have to clean references (and delete obj/bin dirs). .NET Core tends to keep them cached.

**Comment by leppie** [comment]
BTW that is why I have the nupkg in the release too. I should really add it to the github feed, but you can add it to a local feed, so it will pick up the changes.

**Comment by leppie** [comment]
New release, not sure it will make much diff to your issue, but it includes `IronScheme.ConsoleCore.exe` for .NET Core. It will default to what you have installed (probably dropping to lowest version), but works fine for .NET 6.
![image](https://user-images.githubusercontent.com/127666/141533307-b381bed3-4e14-4e4d-bd0c-80ab28289add.png)

**Comment by leppie** [comment]
Fixed with bb23ea5f47476164f8ff7378d74071c2995e8c5c

@bcalco please log a new issue if you can repro with yours
